### PR TITLE
Cypress browser tests

### PR DESCRIPTION
**Pull request by DurkoMatko**
**add**: cypress browser tests for following

- Signup new organizations and users
- Create boxes
- Create products
- General UI navigation (login, logout, change password, invite to organization)

**Couple of details need to be discussed before merging (Friday Hangout?)**

Note to James
- [x] Add ESLint rules
- [ ] Needs to run as part of build

### CODE REVIEW

**Comment by jamescrowley** [comment]
thanks @DurkoMatko ! I'll get the rebased onto the new master. 

What was the main issue that had stopped it working when we last chatted?

**Comment by jamescrowley** [comment]
Note to self: Add ESLint support, and Redux test sample too.

File: src/modules/products/components/ProductDialog.jsx [hunk 1]
```diff
@@ -65,6 +65,7 @@ const ProductDialog = ({
             value={values.category}
             fullWidth
             margin="dense"
+            data-cy="select-category"
```
Comment (jamescrowley) [inline src/modules/products/components/ProductDialog.jsx]: Note to self: let's update to use camelCase like the rest.

File: cypress/support/commands.js [hunk 2]
```diff
@@ -23,3 +23,53 @@
...
+
+Cypress.Commands.add("navigateToInvitePage", () => {
+    cy.openAppDrawer();
+    cy.get("a[data-cy=inviteDrawerButton]").last().click();
+});
+
+//https://github.com/cypress-io/cypress/issues/761
+Cypress.Commands.add("cleanUpXHR", function() {
+    if (Cypress.env("run_all_suite")) {
+      cy.visit(`${Cypress.env("yourAppUrl")}/404`, { failOnStatusCode: false });
```
Comment (jamescrowley) [inline cypress/support/commands.js]: Need to use the correct environment variable

Comment (jamescrowley) [reply cypress/support/commands.js]: Reading the issue, it looks like it's been fixed in 3.1.4? Were you still seeing the problem?

File: cypress/integration/createBoxesAndProducts.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            testPwd = $result.testPwd;
+        });
+    })
+
+    beforeEach(function() {        
+        cy.reLogin(testUserMail, testPwd);
+    });
+
+    afterEach(function() {
+        cy.wait(2000);
```
Comment (jamescrowley) [inline cypress/integration/createBoxesAndProducts.js]: How come?

File: cypress/integration/createBoxesAndProducts.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    });
+
+    it('Create new product', () => {
+        let productName = uuidv4().substring(0,6);
+        cy.navigateToProductsPage();
+        cy.get('button[data-cy=addProductButton]').click();
+        cy.get('#select-category').click();
```
Comment (jamescrowley) [inline cypress/integration/createBoxesAndProducts.js]: I think there's a data-cy added for this?

File: cypress/integration/createBoxesAndProducts.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        cy.get('li[tabindex=0]').click();
```
Comment (jamescrowley) [inline cypress/integration/createBoxesAndProducts.js]: First item in the drop down list or something? Might need to narrow down this selector

Comment (DurkoMatko) [reply cypress/integration/createBoxesAndProducts.js]: Exactly. There's just one dropdown list in the form so I thought it's enough. Or you don't like that he always takes the first element? I should be able to make it random pretty easy if it's encouraged. I just thought it doesn't make much of a difference for the test.

File: cypress/integration/createBoxesAndProducts.js (hunk 4 above)
Comment (DurkoMatko) [reply cypress/integration/createBoxesAndProducts.js]: Yupp, my bad I've forgotten :) Fixed.

File: cypress/integration/createBoxesAndProducts.js (hunk 3 above)
Comment (DurkoMatko) [reply cypress/integration/createBoxesAndProducts.js]: It was just for debugging purposes. Forgotten to delete. Fixed.

File: cypress/support/commands.js (hunk 2 above)
Comment (DurkoMatko) [reply cypress/support/commands.js]: Yeah issue says it's fixed but I still get all the XHRs in previous tests open. But I'm not sure if it's cuz of this issue or it's the issue with Firestore XHRs not being closed we had earlier (starting with @bfirsh mentioning in his initial #292)  I haven't found a solution for it, I just decided to ignore it and implement those tests anyway. XHRs stayed open also in another Git repository with cypress and firestore I've tested.

This `cleanUpXHR` is currently not used, I just kept it in the code for us to discuss during the next call.

File: cypress/integration/controlOperations.js [hunk 6]
```diff
@@ -34,7 +34,7 @@ describe('Control operations', function() {
         });
     });
 
-    it('Change password -> Relogin -> Change password', () => {
+    /*it('Change password -> Relogin -> Change password', () => {
```
Comment (jamescrowley) [inline cypress/integration/controlOperations.js]: Not sure you meant to commit this?

Comment (DurkoMatko) [reply cypress/integration/controlOperations.js]: I commented them out to not waste the recordings allowed for free plan with cypress :D I didn't want to run all tests while triggering the test travis builds :)

File: cypress/integration/createBoxesAndProducts.js (hunk 5 above)
Comment (DurkoMatko) [reply cypress/integration/createBoxesAndProducts.js]: Ok, changed this to `cy.get('li[id=category]').first()` after reading through Cypress documentation

**Pull request merged**